contracts/AbcToken.sol

```diff
@@ -66,9 +66,9 @@ contract AbcToken is Router, ERC20Upgradeable {
         uint32 _destination,
         address _recipient,
         uint256 _amount
-    ) external {
+    ) external payable {
         _burn(msg.sender, _amount);
-        _dispatchToRemoteRouter(_destination, abi.encode(_recipient, _amount));
+        _dispatchWithGasAndCheckpoint(_destination, abi.encode(_recipient, _amount), msg.value);
```

Comment:
should we add a test for paying interchain gas? I could see the argument for not having it, but because the application needs to make sure that payment can be made, it feels worth it?

Reply:
Yes indeed, added